tuf/signed/verify.go

```diff
@@ -107,3 +108,18 @@ func VerifySignature(msg []byte, sig *data.Signature, pk data.PublicKey) error {
 	sig.IsValid = true
 	return nil
 }
+
+// VerifyPublicKeyMatchesPrivateKey checks if the private key and the public keys forms valid key pairs.
+// This should work with both ecdsa-x509 certificate PublicKey as well as ecdsa PublicKey
```

Comment:
Nitpick:  Should this say just "x509 certificate PublicKeys as well as non-certificate PublicKeys"?  (I think RSA keys are also supported, for instance)